packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -297,6 +316,7 @@ function useFocusContainment(scopeRef: RefObject<Element[]>, contain: boolean) {
       if (!activeScope || isAncestorScope(activeScope, scopeRef)) {
         activeScope = scopeRef;
         focusedNode.current = e.target;
+        onSyntheticFocus(e);
```

Comment:
We are in a containing focus scope, which means we need to know whenever an element inside the containing area blurs

on focus, we call onSyntheticFocus, which will use the target of the focus to begin watching the dom for removals of nodes that include our target and will also watch if the target becomes disabled.

when either of those things happens, only chrome will fire a blur event, so we fire our own as well, but only if no previous blur was fired (prevents double blur for chrome). Ours is async, so it'll *always* be after the Chrome native blur event. When blur is called on our target, either because of us or because of the browser, onBlur is called up above where we  setup `onSyntheticFocus` `let onSyntheticFocus = useSyntheticBlurEvent(onBlur);`